### ISSUE TITLE
Implement missing audio, document and image matching functions

### ISSUE THREAD

**Issue by gabriel-vasile** [gabriel-vasile/mimetype#1]
Currently there are several matching functions not implemented, ex: [doc](https://github.com/gabriel-vasile/mimetype/blob/master/matchers/ms_office.go#L22), [xls](https://github.com/gabriel-vasile/mimetype/blob/master/matchers/ms_office.go#L30), [flac](https://github.com/gabriel-vasile/mimetype/blob/master/matchers/audio.go#L10), [wav](https://github.com/gabriel-vasile/mimetype/blob/master/matchers/audio.go#L26). Because of this there are file formats which are not detected by the library.

A good resource for file signatures is https://www.garykessler.net/library/file_sigs.html

**Comment by theotow** [comment]
@gabriel-vasile added a new PR on this, checkout #3

**Comment by gabriel-vasile** [comment]
@WheeskyJack indeed, a list of not yet implemented mimes would be helpful. I will keep this issue open and update it with the latest needed mime types.
For now:

Mime type | Reference | Stub
-------------  | -------------- | ------
application/javascript | https://github.com/file/file/blob/master/magic/Magdir/javascript | [js](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L36)
text/x-lua | https://github.com/file/file/blob/master/magic/Magdir/lua | [lua](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L37)
text/x-perl | https://github.com/file/file/blob/master/magic/Magdir/perl | [perl](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L38)
text/x-python | https://github.com/file/file/blob/master/magic/Magdir/python | [py](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L39)
image/svg+xml | - | [svg](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L41)
model/x3d+xml | - | [x3d](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L42)
application/vnd.google-earth.kml+xml | - | [kml](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L43)
model/vnd.collada+xml | - | [collada](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L44)
application/gml+xml | - | [xml](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L45)
application/gpx+xml | - |[gpx](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L46)

**Comment by TeuryBazzo** [comment]
hi, is someone working on this issue ? if no one  work , i want to try

**Comment by gabriel-vasile** [comment]
Hi, as far as I know no one is working on any of the missing mimetypes. If you decide to have a go, just tell which one you chose.

**Comment by TeuryBazzo** [comment]
I chose text/x-perl, text/x-lua, application/javascript.
I did in local the mimetype application/javascript and text/x-lua, but i used match string with keywords of language (javascript and lua), is the correct way ?

**Comment by TeuryBazzo** [comment]
I chose 

- application/javascript
- text/x-lua
- text/x-perl
- text/x-python

**Comment by TeuryBazzo** [comment]
Hi,
I'd like to do the mime types 
- image/svg+xml
- model/x3d+xml

For this, check the tags of extension .svg and .x3d (<svg, <x3d) is good verification? if not, whats the best way?

**Comment by gabriel-vasile** [comment]
@TeuryBazzo You are on the right track, checking the existence of those tags is the way to detect these subtypes of xml.

For anyone interested, this is the list of mimetypes which need implementation:

Mimetype | Stub
------------- | ------
application/vnd.google-earth.kml+xml | [kml](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L45)
model/vnd.collada+xml | [dae](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L46)
application/gml+xml | [gml](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L47)
application/gpx+xml | [gpx](https://github.com/gabriel-vasile/mimetype/blob/master/tree.go#L48)

The list is not limited, if anyone has a purpose for a mimetype to be added, just ask for it here, or better yet, add a pull request.

**Comment by ahmedsharif865** [comment]
Is someone still working on this issue?

**Comment by gabriel-vasile** [comment]
Hi, the MIME types from the table in my last comment have all been added to the library.
There are still some [ftyp](http://www.ftyps.com/) MIME types to be added, namely: `image/jpx`, `image/jpm`, `image/jp2`. If you want, you are welcome to add them to the library.

**Comment by gabriel-vasile** [comment]
@ahmedsharif865 if you know any other MIME types that should be supported just tell.

**Comment by gabriel-vasile** [comment]
NDJSON support is needed. Details can be found in [issue](https://github.com/gabriel-vasile/mimetype/issues/40). 
JSONL is another format we might add, but I'm not sure if the specification is done and they chose a MIME type for their format.

**Comment by ibraimgm** [comment]
@gabriel-vasile I will take a look at the `image/jpx`, `image/jpm` and `image/jp2` types, if that is ok to you.

**Comment by tebrizetayi** [comment]
@gabriel-vasile  is someone working on the issue? I would like to contribute to this project.

**Comment by EclesioMeloJunior** [comment]
I would like to contribute too! I would know if there is a list of types with missing implementation?

**Comment by gabriel-vasile** [comment]
@EclesioMeloJunior you can add support for linux timezone files or the different formats GIMP uses for graphics.
https://github.com/gabriel-vasile/mimetype/issues/103
https://github.com/gabriel-vasile/mimetype/issues/102

**Comment by fahadsiddiqui** [comment]
Hey, I added OWL file (xml variant) in one of my PR, would like to contribute more. What is up?

**Comment by ibraimgm** [comment]
@gabriel-vasile, PR #128 implements XCF, PAT and GBR.

**Comment by gexaigor** [comment]
Hey, I made support for sxc files. Can I push this?
